File: packages/expo-sensors/android/src/main/java/expo/modules/sensors/services/GravitySensorService.java
// Copyright 2015-present 650 Industries. All rights reserved.

package expo.modules.sensors.services;

import android.content.Context;
import android.hardware.Sensor;

import java.util.Collections;
import java.util.List;

import org.unimodules.core.interfaces.InternalModule;

public class GravitySensorService extends SubscribableSensorService implements InternalModule, org.unimodules.interfaces.sensors.services.GravitySensorService {
  public GravitySensorService(Context reactContext) {
    super(reactContext);
  }

  @Override
  int getSensorType() {
    return Sensor.TYPE_GRAVITY;
  }

  @Override
  public List<Class> getExportedInterfaces() {
    return Collections.<Class>singletonList(org.unimodules.interfaces.sensors.services.GravitySensorService.class);
  }
}

File: packages/expo-sensors/android/src/main/java/expo/modules/sensors/services/GravitySensorService.kt
// Copyright 2015-present 650 Industries. All rights reserved.
package expo.modules.sensors.services

import android.content.Context
import android.hardware.Sensor
import expo.modules.interfaces.sensors.services.GravitySensorServiceInterface
import expo.modules.core.interfaces.InternalModule

class GravitySensorService(reactContext: Context?) : SubscribableSensorService(reactContext), InternalModule, GravitySensorServiceInterface {
  override val sensorType: Int = Sensor.TYPE_GRAVITY

  override fun getExportedInterfaces(): List<Class<*>> {
    return listOf<Class<*>>(GravitySensorServiceInterface::class.java)
  }
}
